src/app/projects/components/project-settings/project-settings.component.ts

```diff
@@ -1,169 +1,138 @@
-  }
-
   constructor(
     private routeService: RouteService,
     private activatedRoute: ActivatedRoute,
     private projectApolloService: ProjectApolloService,
     private router: Router,
     private formBuilder: FormBuilder,
-    private s3Service: S3Service,
-    private informationSourceApolloService: WeakSourceApolloService
-  ) { }
-
-  ngAfterViewInit() {
-    this.inputTaskName.changes.subscribe(() => {
-      this.setFocus(this.inputTaskName);
-    });
-  }
-  setFocus(focusArea) {
-    if (focusArea.length > 0) {
-      focusArea.first.nativeElement.focus();
-    }
+    private s3Service: S3Service
+  ) {
+    this.dataHandlerHelper = new DataHandlerHelper(this.formBuilder, this.projectApolloService);
+    this.lh = new LabelHelper(this, this.projectApolloService);
   }
+
   ngOnInit(): void {
     UserManager.checkUserAndRedirect(this);
     this.routeService.updateActivatedRoute(this.activatedRoute);
-
     const projectId = this.activatedRoute.parent.snapshot.paramMap.get('projectId');
-    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(projectId);
-    this.project$.subscribe((project) => this.project = project);
-
     NotificationService.subscribeToNotification(this, {
       projectId: projectId,
       whitelist: ['tokenization', 'embedding', 'embedding_deleted', 'label_created', 'label_deleted', 'attributes_updated', 'labeling_task_deleted', 'labeling_task_updated', 'labeling_task_created', 'project_update', 'project_export', 'calculate_attribute'],
       func: this.handleWebsocketNotification
     });
     this.setUpCommentRequests(projectId);
-
-    this.requestPKeyCheck(projectId);
     this.checkProjectTokenization(projectId);
 
-    [this.downloadedModelsQuery$, this.downloadedModelsList$] = this.informationSourceApolloService.getModelProviderInfo();
-    this.subscriptions$.push(
-      this.downloadedModelsList$.subscribe((downloadedModels) => this.downloadedModels = downloadedModels));
-
-    let preparationTasks$ = [];
-    preparationTasks$.push(this.prepareAttributesRequest(projectId));
-    preparationTasks$.push(this.prepareLabelingTasksRequest(projectId));
-    preparationTasks$.push(this.prepareEmbeddingsRequest(projectId));
+    [this.projectQuery$, this.project$] = this.projectApolloService.getProjectByIdQuery(projectId);
+    this.subscriptions$.push(this.project$.subscribe((project) => {
+      this.project = project;
+      [this.attributesQuery$, this.attributes$] = this.dataHandlerHelper.prepareAttributesRequest(projectId);
+      [this.embeddingQuery$, this.embeddings$] = this.dataHandlerHelper.prepareEmbeddingsRequest(projectId);
+      this.suggestions$ = this.projectApolloService.getRecommendedEncodersForEmbeddings(projectId);
+      let tasks$ = [];
+      tasks$.push(this.attributes$);
+      tasks$.push(this.embeddings$);
+      tasks$.push(this.suggestions$);
+
+      combineLatest(tasks$).subscribe((res: any[]) => {
```

Comment:
why is this done inside the project subscription? so whenever the project is subscription is setup the attribute & embedding querys are resetup. If we necessarily need the project for the rest we could work with a pipe first but it doesn't seem to be necessary